File: 27.py
# 27. Remove element
# Easy

class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        if not nums:
            return 0

        insert_pos = 0

        for i in range(0,len(nums)):
            if nums[i] != val:
                nums[insert_pos] = nums[i]
                insert_pos += 1

        return insert_pos

File: 27.cpp
// 27. Remove Element
// Easy

class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int insert_pos = 0;

        for (int num : nums)
            if (num != val) nums[insert_pos++] = num;
        return insert_pos;
    }
};
